promlinter.go

```diff
@@ -254,19 +291,44 @@ func (v *visitor) parseCallerExpr(call *ast.CallExpr) ast.Visitor {
 		return v
 	}
 
-	return v.parseOpts(call.Args[0], metricType)
+	return v.parseOpts(call.Args, metricType)
 }
 
-func (v *visitor) parseOpts(optArg ast.Node, metricType dto.MetricType) ast.Visitor {
+func (v *visitor) parseOpts(optArgs []ast.Expr, metricType dto.MetricType) ast.Visitor {
 	// position for the first arg of the CallExpr
-	optsPosition := v.fs.Position(optArg.Pos())
-	opts, help := v.parseOptsExpr(optArg)
+	optsPosition := v.fs.Position(optArgs[0].Pos())
+	opts := v.parseOptsExpr(optArgs[0])
+
+	var metric *dto.Metric
+	if len(optArgs) > 1 {
+		// parse labels
+		if labelOpts := v.parseOptsExpr(optArgs[1]); labelOpts != nil && len(labelOpts.labels) > 0 {
+			metric = &dto.Metric{}
+			for idx, _ := range labelOpts.labels {
+				metric.Label = append(metric.Label,
+					&dto.LabelPair{
+						Name: &labelOpts.labels[idx],
+					})
+			}
+		}
+	}
+
 	if opts == nil {
 		return v
 	}
+
 	currentMetric := dto.MetricFamily{
 		Type: &metricType,
-		Help: help,
+	}
+
+	if !opts.helpSet {
+		currentMetric.Help = nil
```

Comment:
We don't need this condition. We cannot set String to `nil` I guess.
Line 327 should be enough to check if help is set

Reply:
`currentMetric.Help` is type `*string`, we just assign nil to mean "There is no help available". If we set a point to empty string(`""`), some exists test cases will fail.